internal/tescale/scale.go

```diff
@@ -0,0 +1,268 @@
+	v4TunnelIPAddrs := NewIPPool(iputil.GenerateIPs(V4TunnelIPBlock, param.V4TunnelCount))
+	v4VIPAddrs := NewIPPool(iputil.GenerateIPs(V4VIPIPBlock, (param.V4TunnelNHGCount*param.V4TunnelNHGSplitCount)+2))
+	v4EgressIPAddrs := NewIPPool(egressIPs)
+
+	defaultVRF := deviations.DefaultNetworkInstance(dut)
+	vrfTConf := &VRFConfig{Name: VRFT}
+	vrfRConf := &VRFConfig{Name: VRFR}
+	vrfRDConf := &VRFConfig{Name: VRFRD}
+	vrfDefault := &VRFConfig{Name: defaultVRF}
+	idPool := NewIDPool(10000)
+
+	// VRF_T:
+
+	nhgID := idPool.NextNHGID()
+	nhID := idPool.NextNHID()
+	nhgRedirectToVrfR := nhgID
+	// build backup NHG and NH.
+	vrfDefault.NHs = append(vrfDefault.NHs,
+		fluent.NextHopEntry().WithIndex(nhID).WithNetworkInstance(defaultVRF).WithNextHopNetworkInstance(VRFR),
+	)
+	vrfDefault.NHGs = append(vrfDefault.NHGs,
+		fluent.NextHopGroupEntry().WithID(nhgRedirectToVrfR).AddNextHop(nhID, 1).WithNetworkInstance(defaultVRF),
+	)
+
+	// Build IPv4 entry and related NHGs and NHs.
+	// * Mapping tunnel IP per the IP -> NHG ratio
+	// * Each NHG has unique NHs.
+	// * Each NHG has the same backup to Repair VRF.
+	tunnelNHGRatio := param.V4TunnelCount / param.V4TunnelNHGCount
+	for idx, ip := range v4TunnelIPAddrs.AllIPs() {
+		nhgEntry := fluent.NextHopGroupEntry().WithID(nhgID).WithNetworkInstance(defaultVRF).WithBackupNHG(nhgRedirectToVrfR)
+		if idx%tunnelNHGRatio == 0 {
+			nhgID = idPool.NextNHGID()
+
+			// Build NHs and link NHs to NHG.
+			for i := 0; i < param.V4TunnelNHGSplitCount; i++ {
+				vip := v4VIPAddrs.NextIP()
+				nhID = idPool.NextNHID()
+				vrfDefault.NHs = append(vrfDefault.NHs,
+					fluent.NextHopEntry().WithIndex(nhID).WithNetworkInstance(defaultVRF).WithIPAddress(vip),
+				)
+				nhgEntry = nhgEntry.AddNextHop(nhID, 1)
+			}
+			vrfDefault.NHGs = append(vrfDefault.NHGs, nhgEntry)
+		}
+
+		// Build IPv4 entry
+		vrfTConf.V4Entries = append(vrfTConf.V4Entries,
+			fluent.IPv4Entry().WithPrefix(ip).WithNextHopGroup(nhgID).WithNetworkInstance(VRFT).WithNextHopGroupNetworkInstance(defaultVRF),
+		)
+	}
+
+	// Default VRF:
+
+	// * each VIP 1:1 map to a NHG
+	// * each NHG points to unique NHs
+	for _, ip := range v4VIPAddrs.AllIPs() {
+		nhgID := idPool.NextNHGID()
+		nhgEntry := fluent.NextHopGroupEntry().WithID(nhgID).WithNetworkInstance(defaultVRF)
+		// Build NHs and link NHs to NHG.
+		for i := 0; i < param.EgressNHGSplitCount; i++ {
+			vip := v4EgressIPAddrs.AllIPs()[i]
+			nhID = idPool.NextNHID()
+			vrfDefault.NHs = append(vrfDefault.NHs,
+				fluent.NextHopEntry().WithIndex(nhID).WithNetworkInstance(defaultVRF).WithIPAddress(vip),
+			)
+			nhgEntry = nhgEntry.AddNextHop(nhID, 1)
+		}
+
+		vrfDefault.NHGs = append(vrfDefault.NHGs, nhgEntry)
+		// Build IPv4 entry
+		vrfDefault.V4Entries = append(vrfDefault.V4Entries,
+			fluent.IPv4Entry().WithPrefix(ip).WithNextHopGroup(nhgID).WithNetworkInstance(defaultVRF).WithNextHopGroupNetworkInstance(defaultVRF),
+		)
```

Comment:
please set the prefix ( /32 ) for the routes here and other places where Ipv4Entry is created